fuzz/fuzz_atpointer.cpp

```diff
@@ -0,0 +1,68 @@
+#include "simdjson.h"
+#include "FuzzUtils.h"
+#include <cstddef>
+#include <cstdint>
+#include <string>
+#include <string_view>
+
+struct FuzzData {
+    std::string_view json_pointer;
+    std::string_view json_doc;
+};
+
+/**
+ * @brief split split fuzz data into a pointer and a document
+ * @param Data
+ * @param Size
+ * @return
+ */
+FuzzData split(const char *Data, size_t Size) {
+
+    using namespace std::literals;
+    constexpr auto sep="\n~~~\n"sv;
+
+    std::string_view all(Data,Size);
+    auto pos=all.find(sep);
+    if(pos==std::string_view::npos) {
+        //not found.
+        return FuzzData{std::string_view{},all};
+    } else {
+        return FuzzData{std::string_view{all.substr(0,pos)},all.substr(pos+sep.size())};
+    }
+}
+
+extern "C" int LLVMFuzzerTestOneInput(const uint8_t *Data, size_t Size) {
+
+    // Split data into two strings, json pointer and the document string.
+    // Might end up with none, either or both being empty, important for
+    // covering edge cases such as https://github.com/simdjson/simdjson/issues/1142
+    // Inputs missing the separator line will get an empty json pointer
+    // but the all the input put in the document string. This means
+    // test data from other fuzzers that take json input works for this fuzzer
+    // as well.
+    const auto fd=split(as_chars(Data),Size);
+
+    simdjson::dom::parser parser;
+
+    // parse without exceptions, for speed
+    auto res=parser.parse(fd.json_doc.data(),fd.json_doc.size());
```

Comment:
The idiomatic way to do this (in noexcept land) is to use error chaining:

```c++
error_code error = parser.parse(fd.json_doc.data(), fd.json_doc.size()).at_pointer(fd.json_pointer).get_string().error();
if (error) { return 0; }
```

It can be broken up if you want, but just FYI :) Error chaining lets you 

Also, most places we don't store the result and then get it--usually just error_code error = parse.X().get(y).  It's not necessary, but it'll make it easier for people to jump around the codebase. Plus, people will copy/paste/learn from the code anywhere, so making everything use the best idioms seems best :)

Reply:
Ah, nice! I agree with the idea that all code should be treated as if it is example code.
In this case, I am however interested in being able to inspect the intermediate result in a debugger. Plus it looks weird in non-fuzzing code asking for the string but only extract the error?